tests/codegen/vec-in-place.rs

```diff
@@ -37,39 +37,51 @@ pub struct Baz {
 #[no_mangle]
 pub fn vec_iterator_cast_primitive(vec: Vec<i8>) -> Vec<u8> {
     // CHECK-NOT: loop
-    // CHECK-NOT: call
```

Comment:
Those checks are asserting important properties (no allocator API calls, no inlining failures).
If possible they should be refined instead.

Reply:
I have updated the checks. However I have to introduce a very complex regular expression in `tests/codegen/vec_pop_push_noop.rs` for all different configurations, because the regular expression engine used by FileCheck doesn’t support lookahead. That check means “if there are any `call` statements, they will have to be `call void @llvm.assume` statements”.